IPython/html/static/widgets/js/widget_bool.js

```diff
@@ -128,11 +128,12 @@ define([
                 this.$el.prop('disabled', disabled);
 
                 var description = this.model.get('description');
+                var faclass = this.model.get("icon");
                 this.$el.attr("title", this.model.get("tooltip"));
-                if (description.trim().length === 0) {
+                this.$el
+                   .html('<i class="fa '.concat(faclass, '"></i>', description));
```

Comment:
.html calls should never take a user-defined value. Always use $.text. Also need to make sure that faclass. This adds two separate places for users to create incorrect HTML (faclass and description).